Cargo.toml

```diff
@@ -10,4 +10,6 @@ license = "Apache-2.0 OR BSD-3-Clause"
 edition = "2018"
 
 [dependencies]
-libc = ">=0.2.98"
+libc = ">=0.2.39"
+serde = { version = ">=1.0.27", features = ["derive"] }
+serde_json = ">=1.0.9"
```

Comment:
It looks like for now the json frontend is not going to be used by all the consumers of seccompiler. For this use case, what do you think about gating this functionality under a feature? We could also consider converting this to a workspace and offer seccompiler and seccompiler-json as 2 separate crates as well. The later is preferred because then users don't need to import serde at all (not even in the manifest). This would make possible by using the serde functionality of remotes: https://serde.rs/remote-derive.html.

The reason I'm saying that separating it in 2 crates is preferred is because you can still hit incompatibilities that make your project un-buildable if the version of serde used in this crate is incompatible with the version of serde used in the consumer project. This happens even when the serde dependency is hidden behind a feature that is unused in the consumer project.

Reply:
Okaay, so my information was outdated. It seems that cargo does the right thing with optional dependencies now. So if you are not using the json feature, then you won't have serde as a dependency. My personal preference is still to separate the JSON implementation from the basic one, but since there is no associated problem with this, it remains just that - a preference. Resolving this.